### PR TITLE
feat: add types

### PR DESCRIPTION
**Pull request by wisepotato**
* Add type hints
* Add to Changelog
* Remove old type hints as they are not relevant for 3.5+ (is this ok? I've never seen these type hints, i wonder if they are for 2.*?)

### CODE REVIEW

**Comment by ivankorobkov** [comment]
Hi, thanks for the PR. Please, update the .travis.yml and make all the tests pass :-) See https://travis-ci.org/ivankorobkov/python-inject/builds/613510358?utm_source=github_status&utm_medium=notification

**Comment by wisepotato** [comment]
will do!

**Comment by wisepotato** [comment]
done @ivankorobkov, are you able to deploy this for v4/ does it conform to what you expect?

**Comment by ivankorobkov** [comment]
Hi! Thanks for your contribution. Everything seems fine. I'll make the next release on the weekend.

File: inject/__init__.py [hunk 1]
```diff
@@ -382,11 +246,142 @@ def __call__(self, func):
...
+
+    with _INJECTOR_LOCK:
+        if _INJECTOR is None:
+            return
+
+        _INJECTOR = None
+        logger.debug('Cleared an injector')
+
+
+def instance(cls: T) -> T:
```
Comment (Enforcer) [inline inject/__init__.py]: @wisepotato why was this (and similar functions, e.g. `attr`) annotated to return the same type of argument as the that was given as argument?

I would expect to give there a class (type) as an argument and get a class instance in return, so in my opinion, it should look like this:
```python
def instance(cls: Type[T]) -> T:
    ...
```

This was a version I originally added in .pyi: https://github.com/ivankorobkov/python-inject/pull/21/files#diff-ebdc46604c1a2a1ca81ca20331a18fe5L26

Comment (ivankorobkov) [reply inject/__init__.py]: Hi, I think, it's just a typo.

Comment (wisepotato) [reply inject/__init__.py]: Does this differ in Python? I did not know about `Type[T]` and `T`, any reference on that?

**Pull request merged**